local/local_test.go

```diff
@@ -189,6 +189,35 @@ func testImage(t *testing.T, when spec.G, it spec.S) {
 		})
 	})
 
+	when("#Labels", func() {
+		when("image exists", func() {
```

Comment:
Do we want a test case for when the image does not exist? Or when there are no labels? I think right now both of these scenarios return `nil`. This opens the caller up to more potential errors than returning an empty map would, but is probably accurate  given that the image config json has `nil` rather than `{}`, `labels` in these cases.

Instead of changing `img.Labels()` to `return map[string]string` when the the config file has `nil` labels I think we should change our empty image such that labels are never `nil`.

Reply:
I added test for missing image and when there are no labels on an existing image.

Isn't `nil` eq to an empty map in this case? The tests I wrote seem to imply there is no functional difference between the two. 

I know `nil` or empty maps are not writeable, is that the concern? Do you think we need to `make(map[string]string)`?

Reply:
Yeah you are right that the circumstances where this would become an issue are a bit esoteric. `SetLabel` works without errors because we initialize an empty map before adding a key if the labels are `nil`. This would only become an issue if you you called `img.Label()` and then tried set a key on the returned value (rather than calling `SetLabel` on the image).

I am fine leaving local/remote as is but let's change the fake to match the implementations just to be cautious.

Reply:
Alright, I pushed up a change to `SetLabel` for the fake implementation. I think that is what you meant for me to do..right?

Reply:
Yes we need that change in `SetLabel`. Also, I think we should change `fakes.NewImage` to initialize the image with `nil` labels instead of empty labels https://github.com/buildpacks/imgutil/blob/main/fakes/image.go#L23

Then the behavior of `Labels` will be the same for fake, local, and remote images in cases where there are no labels.

Reply:
Ahh, ok. I see.